tests/countries/test_niger.py

```diff
@@ -0,0 +1,336 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays import OPTIONAL
+from holidays.countries.niger import Niger, NE, NER
+from tests.common import CommonCountryTests
+
+
+class TestNiger(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1960, 2050)
+        super().setUpClass(Niger, years=years, years_non_observed=years)
+        cls.no_estimated_holidays = Niger(years=years, islamic_show_estimated=False)
+        cls.optional_holidays = Niger(categories=OPTIONAL, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(Niger, NE, NER)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Niger(years=1958))
+
+    def test_new_years_day(self):
+        name = "Jour de l'An"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1998, 2050)))
+        obs_dt = (
+            "2012-01-02",
+            "2017-01-02",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observé)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+        self.assertNoHolidayName(f"{name} (observé)", range(1960, 1998))
```

Comment:
We have not used such check before, but it seems useful. 👍 It's only advisable to use `name_observed = f"{name} (observé)"` variable.